### ISSUE TITLE
Isochrones responding with weird squares

### ISSUE THREAD

**Issue by nilsnolde** [valhalla/valhalla#4459]
Ref https://github.com/gis-ops/valhalla-qgis-plugin/issues/17

Somehow isochrones can produce those squares. I guess it's basically no reachability and the contouring spits out a single square?! Anyways, should be easy to reproduce.

**Comment by chrstnbwnkl** [comment]
Seems like in that specific neighborhood of Fortaleza, there are a lot of ways that have pretty restrictive `maxheight`, `maxwidth` and `maxlength` tags: https://www.openstreetmap.org/way/278728789. The `maxlength` and `maxwidth` values are below the default values of Valhalla's `length` and `width` parameters for truck costing. When doing the isochrone expansion, the grid cells that contain a request location are automatically set: 

https://github.com/valhalla/valhalla/blob/efff65d35c68c9198d504e07087d3ae2b71e5e1f/src/thor/isochrone.cc#L165C1-L171C2 

Which produces these weird square shapes. The issue can be [reproduced in the web app](https://valhalla.openstreetmap.de/isochrones?profile=truck&wps=-38.502615774067856%2C-3.7390036438792738&range=10&interval=10&generalize=0&denoise=0.1
).

I have come across these square shapes a couple of times before, they can also occur when setting a location outside of the used graph extract in some contexts.

I guess this sort of result is difficult to interpret without knowing about the underlying method used to produce the contours... an easy way to fix this would be to simply keep track of the number of grid cells whose values are set below the initial max value and return an error if that number is equal to or below the number of specified locations.

**Comment by nilsnolde** [comment]
Thanks for the insight here! So it’s really just single cell being contoured? The proposed fix sounds good. Do even return an error when we can’t produce an isochrone or just return an empty geometry? I think we should do the latter, actually nothing is really wrong in terms of user input or processing, it’s just not able to expand anywhere, which IMO is a 200 status still. But that’s more philosophical, I’m not very convinced about either myself:) 

what do others think?

**Comment by chrstnbwnkl** [comment]
Yep fair enough. Maybe add a warning to the empty response? That way the user understands what's going on

**Comment by nilsnolde** [comment]
> However, other areas of the grid generate isochrones that advance perfectly in front of the avenues that contain these restrictive parameters.

if that's happening, it'd be a bug though. can you share a request to exemplify?

**Comment by chrstnbwnkl** [comment]
Yeah that would be interesting indeed. I tried more examples using our web app, but couldn't reproduce non-square truck isochrones when snapped to edges with these strict max dimension values

**Comment by nilsnolde** [comment]
You might want to experiment with the “denoise” parameter, that removes any inner/outer rings which are under a certain percentage of the biggest outer ring. But also we just shipped a PR which probably helps fixing some issues with polygons in general, I’ll put that live now on the FOSSGIS instance.

Plus I wouldn’t be _too_ confident the QGIS plug-in is actually doing all the right thing either with those polygons, it’s been a bit messed up in Valhalla until a few hours ago😅
